desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLayoutTransformer.java

```diff
@@ -188,6 +188,13 @@ private Point createPoint(final Object vertex, int xIndex, boolean onlyIfCoordin
         if (vertex instanceof Table) {
             JobGraphMetadata.setPointForTable(_analysisJobBuilder, (Table) vertex, point.x, point.y);
         }
+        
+        if (point.x < 0 || point.y < 0) {
+            // apply max(offset, value) to avoid location outside of canvas
+            final int xValue = Math.max(X_OFFSET, point.x);
+            final int yValue = Math.max(Y_OFFSET, point.y);
+            point = new Point(xValue, yValue);
+        }
```

Comment:
It only takes into account negative coordinates. I guess the positive limits should not be there, but we need to fix #146 then ... hmm I guess I can add a limit to that for now also.